core/src/main/java/org/infinispan/manager/impl/ReplicableManagerFunctionCommand.java

```diff
@@ -61,6 +61,7 @@ public boolean isReturnValueExpected() {
 
    @Override
    public boolean canBlock() {
+      // These commands can be arbitrary user commands - so be careful about them blocking
```

Comment:
I was hoping that the result of `canBlock()` would already be ignored, and it would be the command's job to submit a task to the blocking executor if needed. Maybe in the next version, then :)

Reply:
Hrmm, I think I understand now. You mean inside the `invokeAsync` directly. I can make a new JIRA to do that as it will inflate this probably a bit.

Reply:
https://issues.redhat.com/browse/ISPN-11483